provisioning/transport/http/src/Generated/RuntimeRegistration.cs

```diff
@@ -484,135 +456,121 @@ public RuntimeRegistration(DeviceProvisioningServiceRuntimeClient client)
                     { "idScope", idScope },
                     { "cancellationToken", cancellationToken }
                 };
-                ServiceClientTracing.Enter(_invocationId, this, "RegisterDevice", tracingParameters);
+                ServiceClientTracing.Enter(invocationId, this, "RegisterDevice", tracingParameters);
             }
+
             // Construct URL
             string baseUrl = Client.BaseUri.AbsoluteUri;
             string url = new Uri(
                     new Uri(baseUrl + (baseUrl.EndsWith("/", StringComparison.Ordinal) ? "" : "/")),
-                    "{idScope}/registrations/{registrationId}/register")
+                    $"{WebUtility.UrlEncode(idScope)}/registrations/{WebUtility.UrlEncode(registrationId)}/register")
```

Comment:
I'm assuming this is the bug fix, right? Everything else is just code style fixes?

Reply:
And why do we need to URL encode the id scope? It is alphanumeric, right?

Reply:
I retained the behavior of encoding it. I agree we could make the assumption that ID scopes don't include escapable characters, but that doesn't mean a customer couldn't pass bad data. I figured it would be better to continue to encode it (and decode for this path) so the error the user gets is perhaps more meaningful.